### PR TITLE
Add a new_cluster function to xcube.util.dask

### LINKED ISSUE #813: Add a new_cluster function to xcube.util.dask
**Is your feature request related to a problem? Please describe.**

There is no simple and standard way for a user to create a dask cluster in an xcube environment.

**Describe the solution you'd like**

The `new_cluster` function from the Agriculture Virtual Laboratory project should be adapted and integrated into `xcube.util.dask`.

### PR DESCRIPTION
**Pull request by pont-us**
Fixes #813 .

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [x] New/modified features documented in `docs/source/*`
* [x] Changes documented in `CHANGES.md`
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ dependencies:
   - affine >=2.2
   - click >=8.0
   - cmocean >=2.0
+  - coiled >=0.3.1
```
Comment (forman) [inline environment.yml:12]: No. `coiled` is optional.

File: xcube/util/dask.py [hunk 2]
```diff
@@ -135,6 +139,67 @@ def get_chunk_slices(chunk_sizes: Sequence[int]) -> Iterable[slice]:
...
+            'cost-center': 'unknown',
+            'environment': 'dev',
+            'creator': 'auto',
+            'purpose': 'xcube dask cluster',
+        }
+    # TODO: there should be a way to set account and tags without passing them
+    # as function arguments. Reading them from environment variables, perhaps.
+
+    if provider == 'coiled':
+        import coiled
```
Comment (forman) [inline xcube/util/dask.py:162]: Catch `ImportError` and re-raise with message saying that provide "coild" requires package `coiled` to be installed.

Comment (forman) [inline xcube/util/dask.py:162]:
```suggestion
            try:
                import coiled
            except ImportError as e:
                raise ImportError(f"provider 'coiled' requires package 'coiled' to be installed") from e            
```

**Review by forman — CHANGES_REQUESTED** [review]

File: environment.yml (hunk 1 above)
Comment (pont-us) [reply environment.yml]: Fixed.

**Comment by forman** [comment]
@pont-us can you please think of a way to unit-test the new function. Can we mock the `coiled` client interface?

**Comment by pont-us** [comment]
> @pont-us can you please think of a way to unit-test the new function. Can we mock the `coiled` client interface?

Shouldn't be too hard, looking into it now.

File: xcube/util/dask.py [hunk 3]
```diff
@@ -12,6 +17,8 @@
 IntTupleIterable = Iterable[IntTuple]
 SliceTupleIterable = Iterable[SliceTuple]
 
+_CLUSTER_TAGS_ENV_VAR_NAME = 'XCUBE_CLUSTER_TAGS'
+_CLUSTER_ACCOUNT_ENV_VAR_NAME = 'XCUBE_CLUSTER_ACCOUNT'
```
Comment (forman) [inline xcube/util/dask.py:21]:
```suggestion
_CLUSTER_TAGS_ENV_VAR_NAME = 'XCUBE_DASK_CLUSTER_TAGS'
_CLUSTER_ACCOUNT_ENV_VAR_NAME = 'XCUBE_DASK_CLUSTER_ACCOUNT'
```

File: xcube/util/dask.py [hunk 4]
```diff
@@ -135,6 +142,121 @@ def get_chunk_slices(chunk_sizes: Sequence[int]) -> Iterable[slice]:
...
+    n_workers: int = 4,
+    resource_tags: Optional[Dict[str, str]] = None,
+    account: str = None,
+    **kwargs,
+) -> distributed.deploy.Cluster:
+    """
+    Create a new Dask cluster
+
+    Cloud resource tags can be specified in an environment variable
+    XCUBE_CLUSTER_TAGS in the format
```
Comment (forman) [inline xcube/util/dask.py:158]:
```suggestion
    XCUBE_DASK_CLUSTER_TAGS in the format
```

File: xcube/util/dask.py [hunk 5]
```diff
@@ -135,6 +142,121 @@ def get_chunk_slices(chunk_sizes: Sequence[int]) -> Iterable[slice]:
...
+    ``tag_1=value_1:tag_2=value_2:...:tag_n=value_n``. In case of
+    conflicts, tags specified in ``resource_tags`` will override tags
+    specified by the environment variable.
+
+    The cluster provider account name can be specified in an environment
+    variable ``XCUBE_CLUSTER_ACCOUNT``. If the ``account`` argument is
```
Comment (forman) [inline xcube/util/dask.py:164]:
```suggestion
    variable ``XCUBE_DASK_CLUSTER_ACCOUNT``. If the ``account`` argument is
```

**Review by forman — COMMENTED** [review]
Perfect, thanks :)

However, I still have a preference for `XCUBE_DASK_CLUSTER_...` instead of just `XCUBE_CLUSTER_...`, to separate from all other cluster-stuff that might exist in the xcube context, for example clustering classification algorithms or clusters of xcube servers..

**Pull request merged**